scripts/build/builders/nxp.py

```diff
@@ -240,6 +240,9 @@ def generate(self):
             if 'ZEPHYR_NXP_BASE' not in os.environ:
                 raise Exception("ZEPHYR_NXP_BASE need to be set")
 
+            if 'ZEPHYR_TOOLCHAIN_VARIANT' in os.environ:
```

Comment:
@andy31415 FYI, this fixes the nxp zephyr examples in the vscode image. This variable is set by nrf env, and clashes with our own, so we just unset it.

Reply:
Note however that the intent of the build scripts is that I can do a `--dry-run` and see shell commands required to execute the build. So you are not allowed to do run python specific commands: in this case del of the environment (or in other cases file manipulation or python process execution).

Please add the logic to the `cmd` to be executed by the shell, even if ugly.

Reply:
Understood, I changed it by adding `unset ZEPHYR_TOOLCHAIN_VARIANT` in `cmd`.